### PR TITLE
test(range): migrate tests to playwright

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): Tech debt

## What is the current behavior?

Tests are written in puppeteer. 

Issue URL: N/A

## What is the new behavior?

- Migrates tests to Playwright
- Adds new matcher for `toHaveReceivedEventDetail`
- Removes duplicate RTL tests

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/range/test/basic/range.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import { expect } from '@playwright/test';
+import { dragElementBy, test } from '@utils/test/playwright';
+
+test.describe('range: basic', () => {
+  test('should not have visual regressions', async ({ page }) => {
+    await page.goto(`/src/components/range/test/basic`);
+
+    await page.setIonViewport();
+
+    expect(await page.screenshot({ fullPage: true })).toMatchSnapshot(`range-diff-${page.getSnapshotSettings()}.png`);
```
Comment (liamdebeasi) [inline core/src/components/range/test/basic/range.e2e.ts:10]: If you do `setIonViewport`, do we even need to do `fullPage: true`?

Comment (liamdebeasi) [reply core/src/components/range/test/basic/range.e2e.ts:10]: same question goes for other instances of this in the PR

**Review by liamdebeasi — APPROVED** [review]
Great job! My open comment is not a blocker.

File: core/src/components/range/test/basic/range.e2e.ts (hunk 1 above)
Comment (sean-perkins) [reply core/src/components/range/test/basic/range.e2e.ts]: Yep, good catch! Github Copilot auto completed those sections. They aren't necessary when the test template uses `ion-content` + `setIonViewport()`. I will remove 👍

**Review by amandaejohnston — APPROVED** [review]

**Pull request merged**